hare3/graded_gossip_test.go

```diff
@@ -0,0 +1,41 @@
+package hare3
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGradedGossip(t *testing.T) {
+	gg := NewDefaultGradedGossiper()
```

Comment:
please use table tests, or named subtests with extracted function like you mentioned previously

Reply:
The reason I didn't do this here is I think its quite clear with the comment what is happening at each point, and when the test fails you will get a line number taking you to the exact point of the failure. I think the comments are too long to serve as good subtest names, so by adding subtests you'll end up with lots of tests with largely meaningless names which are adding boilerplate to the test. I don't really see what you'll gain from doing that?